tests/unit/abort_source_test.cc

```diff
@@ -161,3 +161,14 @@ SEASTAR_THREAD_TEST_CASE(test_sleep_abortable_with_exception) {
     }
     BOOST_REQUIRE(caught_exception);
 }
+
+SEASTAR_TEST_CASE(test_destroy_subscribed_abort_source) {
```

Comment:
Why is abort-source test here? The abort_source doesn't have any gates inside.

Reply:
(Relevant for https://github.com/scylladb/scylladb/issues/12183)